components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -296,6 +296,7 @@ private String getIDToken(String clientId, String spTenantDomain, JWTClaimsSet j
                               String signingTenantDomain) throws IdentityOAuth2Exception {
 
         if (oAuthAppDO.isIdTokenEncryptionEnabled()) {
+            checkIfPublicCertConfigured(clientId, spTenantDomain);
```

Comment:
Better to return the availability of the jwks url/certificate from this method and throw the exception here(inside getIDToken)

Reply:
Tricky to do that since we throw an exception in the `OAuth2Util.getX509CertOfOAuthApp()` method if a certificate is not configured.